compiler/src/Language/Bond/Codegen/Java/SerializationMethods.hs

```diff
@@ -278,3 +280,13 @@ readMap java keyType valueType fieldName depth =
         countLocal = "count" ++ show depth
         keyLocal = "k" ++ show depth
         valueLocal = "v" ++ show depth
+
+readBonded :: MappingContext -> Type -> String -> Text
+readBonded java bondedType fieldName =
```

Comment:
If I'm reading the implementation, `readBonded` is a partial function over `Type` that only works when the `Type` is `BT_Bonded`. However, `readBonded` is written as if it were total.